sdk/program/src/vote/state/vote_state_deserialize.rs

```diff
@@ -99,16 +101,28 @@ fn read_prior_voters_into<T: AsRef<[u8]>>(
     // Safety: if vote_state is non-null, prior_voters is guaranteed to be valid too
     unsafe {
         let prior_voters = addr_of_mut!((*vote_state).prior_voters);
-        let prior_voters_buf = addr_of_mut!((*prior_voters).buf) as *mut (Pubkey, Epoch, Epoch);
-
-        for i in 0..MAX_ITEMS {
-            let prior_voter = read_pubkey(cursor)?;
-            let from_epoch = read_u64(cursor)?;
-            let until_epoch = read_u64(cursor)?;
-
-            prior_voters_buf
-                .add(i)
-                .write((prior_voter, from_epoch, until_epoch));
+        let prior_voters_buf = addr_of_mut!((*prior_voters).buf) as *mut PriorVotersItem;
+
+        if cfg!(target_endian = "little") {
+            let total_bytes = MAX_ITEMS.saturating_mul(std::mem::size_of::<PriorVotersItem>());
+            let buffer_ptr = prior_voters_buf as *mut u8;
+
+            // Read the bytes directly into the buffer's uninitialized memory
```

Comment:
nit: would be good to say here that bincode by default serializes as little
endian (I had to go and double check the spec)